amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -387,6 +412,77 @@ void runProcessLoop() {
+
+            // This is done to ensure that we clean up the stale streams lingering in the lease table.
+            syncStreamsFromLeaseTableOnAppInit();
+
+            for (StreamIdentifier streamIdentifier : newStreamConfigMap.keySet()) {
+                if (!currentStreamConfigMap.containsKey(streamIdentifier)) {
+                    log.info("Found new stream to process: " + streamIdentifier + ". Syncing shards of that stream.");
+                    ShardSyncTaskManager shardSyncTaskManager = createOrGetShardSyncTaskManager(newStreamConfigMap.get(streamIdentifier));
+                    shardSyncTaskManager.syncShardAndLeaseInfo();
+                    currentStreamConfigMap.put(streamIdentifier, newStreamConfigMap.get(streamIdentifier));
+                    streamsSynced.add(streamIdentifier);
+                } else {
+                    if (log.isDebugEnabled()) {
+                        log.debug(streamIdentifier + " is already being processed - skipping shard sync.");
+                    }
+                }
+            }
+
+            // TODO: Remove assumption that each Worker gets the full list of streams
+            Iterator<StreamIdentifier> currentStreamConfigIter = currentStreamConfigMap.keySet().iterator();
+            while (currentStreamConfigIter.hasNext()) {
+                StreamIdentifier streamIdentifier = currentStreamConfigIter.next();
+                if (!newStreamConfigMap.containsKey(streamIdentifier)) {
+                    log.info("Found old/deleted stream: " + streamIdentifier + ". Syncing shards of that stream.");
+                    ShardSyncTaskManager shardSyncTaskManager = createOrGetShardSyncTaskManager(currentStreamConfigMap.get(streamIdentifier));
+                    shardSyncTaskManager.syncShardAndLeaseInfo();
+                    currentStreamConfigIter.remove();
+                    streamsSynced.add(streamIdentifier);
+                }
+            }
+            streamSyncWatch.reset().start();
+        }
+        return streamsSynced;
+    }
+
+    private Set<StreamIdentifier> syncStreamsFromLeaseTableOnAppInit()
+            throws DependencyException, ProvisionedThroughputException, InvalidStateException {
+        if (!leasesSyncedOnAppInit && isMultiStreamMode) {
+            final Set<StreamIdentifier> streamIdentifiers = leaseCoordinator.leaseRefresher().listLeases().stream()
+                    .map(lease -> StreamIdentifier.multiStreamInstance(((MultiStreamLease) lease).streamIdentifier()))
+                    .collect(Collectors.toSet());
+            for (StreamIdentifier streamIdentifier : streamIdentifiers) {
+                if (!currentStreamConfigMap.containsKey(streamIdentifier)) {
+                    currentStreamConfigMap.put(streamIdentifier, getDefaultStreamConfig(streamIdentifier));
+                }
+            }
+            leasesSyncedOnAppInit = true;
+        }
+        return Collections.emptySet();
```

Comment:
Explanation on why this is returning emptyset would have more context. ?

Reply:
Realized that after refactoring, we no longer need to return anything. Thanks for catching.